esp/services/ws_access/ws_accessService.cpp

```diff
@@ -2315,9 +2315,33 @@ bool Cws_accessEx::onAddViewColumn(IEspContext &context, IEspAddViewColumnReques
 
         checkUser(context);
 
+        const char* origFileName = req.getFilename();
+        const char* origColumnName = req.getColumnname();
+
+        int fileLength = strlen(origFileName);
+        int columnLength = strlen(origColumnName);
+
+        if (fileLength == 0 || columnLength ==0)
+        {
+            resp.setSuccess(false);
+            resp.setResult("Filename or Columnname cannot be empty");
+            return false;
+        }
+
+        // All view column filename MUST be a full path including the scope, with the leading tilde (~)
+        // If filename doesn't start with a leading tilde, add it
+        StringBuffer normalizedFileName;
+
+        if (origFileName[0] != '~')
+        {
+            normalizedFileName.append('~'); 
```

Comment:
When an admin is adding a view, admin will be fetching file/column names from dfu esp service, and dfu returns full paths for each logical file including the scope name. 

However, these filenames do not include leading tildes for some reason (scope::folder::file), so I am adding these checks to be 100% sure that filenames we are adding to LDAP have leading tildes in it (~scope::folder::file).

The problem of filenames not including scope name only arises from field usage within compiled queries, and I will be adding appropriate checks for it in HPCC-15180.